raiden-ts/src/transport/epics/helpers.ts

```diff
@@ -145,12 +150,12 @@ export function waitMemberAndSend$<C extends { msgtype: string; body: string }>(
       // up to RETRY_COUNT times; if it continues to error, throws down
       err$.pipe(
         withLatestFrom(config$),
-        mergeMap(([err, { pollingInterval }], i) => {
-          if (i < RETRY_COUNT - 1) {
-            log.warn(`messageSend error, retrying ${i + 1}/${RETRY_COUNT}`, err);
+        mergeMap(([err, { pollingInterval }], count) => {
+          // always retry rate-limit errors
+          if (count < RETRY_COUNT - 1 || err?.httpStatus === 429) {
+            log.warn(`messageSend error, retrying ${count + 1}/${RETRY_COUNT}`, err);
```

Comment:
Here you don't use `retryWaitWhile` because you want to log something, right?

Reply:
Yes, exactly. We could look into adding some message to be logged to `retryWaitWhile`/`retryAsync$`, but since this isn't that common, I didn't do it yet on this PR